### PR TITLE
Bug 2100439: Enable GCP PD in-tree tests

### PR DESCRIPTION
**Pull request by bertinatto**
Those tests have been disabled by default in k8s 1.24. See: https://github.com/kubernetes/kubernetes/pull/109541

CC @openshift/storage

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,11 @@
 all: build
 .PHONY: all
 
+# Since k8s 1.24 this is required to enable GCP PD tests.
+# Here it's mostly used by update targets. See:
+# https://github.com/kubernetes/kubernetes/pull/109541
+export ENABLE_STORAGE_GCE_PD_DRIVER := yes
```
Comment (bertinatto) [inline Makefile:7]: This doesn't look like the right place for setting this env var, please advise.

Comment (bertinatto) [reply Makefile:7]: CC @bparees 
Any ideas?

Comment (bparees) [reply Makefile:7]: you can set it here, to put it a bit closer to where it matters:

https://github.com/openshift/origin/blob/6438b04e175f9e2ce434b7deb4b786f602eb886b/hack/update-generated.sh#L6

e.g.:
```
ENABLE_STORAGE_GCE_PD_DRIVER=yes go generate -mod vendor ./test/extended
```
should do it.

the bigger problem is you also need to set this in the actual CI jobs where you want this test to run.

File: hack/update-generated.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@ STARTTIME=$(date +%s)
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
 # Update test names
-go generate -mod vendor ./test/extended
+ENABLE_STORAGE_GCE_PD_DRIVER=yes go generate -mod vendor ./test/extended
```
Comment (bparees) [inline hack/update-generated.sh:6]: include a comment/link about why this is here.

Comment (bertinatto) [reply hack/update-generated.sh:8]: Added comment back.

**Pull request merged**